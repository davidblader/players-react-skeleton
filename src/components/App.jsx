import React from 'react';
import { BrowserRouter as Router, Route, Switch } from 'react-router-dom';
import { Transition, CSSTransition } from 'react-transition-group';
import Homepage from './Homepage';
import Login from './Login';
import Registration from './Registration';
import Roster from './Roster';
import NewPlayer from './NewPlayer';
import NewVillager from './NewVillager';

export const setSession = (JWT, user) => {
  localStorage.setItem('JWT', JWT);
  localStorage.setItem('user', JSON.stringify(user));
};

export const logout = () => {
  localStorage.removeItem('JWT');
  localStorage.removeItem('user');
  window.location.replace('/');
};

const App = () =>
  (
    <Router >
      <Switch>
        <Route exact path="/" component={Homepage} />
        <Route path="/login" render={routerProps => <Login {...routerProps} setSession={setSession} />} />
        <Route path="/register" render={routerProps => <Registration {...routerProps} setSession={setSession} />} />
        <Route path="/roster" render={routerProps => <Roster {...routerProps} logout={logout} />} />
        <Route path="/player/new" component={NewPlayer} />
        <Route path="/player/existing" component={NewVillager} />
      </Switch>
    </Router>
  );

export default App;
