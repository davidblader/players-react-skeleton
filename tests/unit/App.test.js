import React from 'react';
import renderer from 'react-test-renderer';
import App, { setSession, logout } from '../../src/components/App';

test('App loads & matches snapshot', () => {
  const appComponent = renderer.create(<App />);
  expect(appComponent.toJSON()).toMatchSnapshot();
});

test('setSession function populates JWT and user object in localStorage', () => {
  const testJWT = 'fakeJWT1234567890';
  const user = {
    id: 'userid1234',
    first_name: 'test',
    last_name: 'user',
    email: 'test@user.com',
  };
  setSession(testJWT, user);
  expect(localStorage.getItem('JWT')).toBe(testJWT);
  expect(JSON.parse(localStorage.getItem('user'))).toEqual(user);
});

test('logout function clears localStorage items', () => {
  logout();
  expect(localStorage.getItem('JWT')).toBeNull();
  expect(localStorage.getItem('user')).toBeNull();
  expect(window.location.pathname).toBe('/');
});
