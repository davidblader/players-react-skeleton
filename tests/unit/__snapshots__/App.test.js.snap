// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App loads & matches snapshot 1`] = `
<div
  className="animal-crossing-box"
>
  <span
    className="animal-crossing-header"
  >
    Animal Crossing Fantasy Village
  </span>
  <div
    className="btn-container"
  >
    <a
      href="/login"
      onClick={[Function]}
    >
      <button
        className="animal-crossing-btn"
      >
        Login
      </button>
    </a>
  </div>
  <div
    className="btn-container"
  >
    <a
      href="/register"
      onClick={[Function]}
    >
      <button
        className="animal-crossing-btn"
      >
        Register
      </button>
    </a>
  </div>
</div>
`;
